text/0035-schema-applies-to-unspecified.md

```diff
@@ -0,0 +1,122 @@
+
+These changes remove some of the confusing behavior of the current implementation (e.g., setting only one of `appliesTo.principalTypes`/`appliesTo.resourceTypes` to be the empty list), while maintaining backwards compatibility.
+Under this alternative, some schemas will produce parse errors after upgrading to Y, but users can fix these error proactively.
+For every valid schema in Y, there is some schema in X that has identical behavior.
+
+The downside of this alternative is that it still allows the empty list to mean that an action applies to no entities, which may be confusing for users.
+Specifying that an action applies to no entities also requires more characters with this alternative, possibly making it more annoying to type out.
+
+This alternative will require updates to [RFC 24](https://github.com/cedar-policy/rfcs/blob/main/text/0024-schema-syntax.md) in order to maintain consistency between the two schema formats.
+
+### Alternative B
+
+Another way to address backwards compatibility concerns is to combine this change with an implementation of schema versioning (see [RFC 33](https://github.com/cedar-policy/rfcs/pull/33)), supporting both the old and new behavior for schema parsing (for some period of time), allowing users to choose which behavior they want by setting a flag.
+
+There are at least two reasonable options here:
+
+- Force Cedar Y schemas to include an explicit version marker saying version Y. Schemas with no version marker will be interpreted using the Cedar X behavior, even if you're using Cedar Y. This gives full backwards-compatibility: no one's schema changes behavior when upgrading from Cedar X to Cedar Y, unless/until you add a version marker saying version Y. However, this means the behavior proposed in this RFC won't be default, which seems undesirable.
+- Same thing, but interpret schemas with no version marker using the Cedar Y behavior. This is still a breaking change, because unmarked schemas will change behavior when upgrading from Cedar X to Cedar Y. But it solves the problem of "no way to write a schema so that validation behavior will be the same between X and Y." Users just add a "version X" marker to their schema, and then they can upgrade from Cedar X to Cedar Y without that schema changing behavior.
+
+The primary arguments against this alternative are that (1) it blocks this RFC on the acceptance of RFC 33, and (2) it commits us to maintaining multiple versions of the schema parser and sets a precedent for supporting deprecated data formats going forward.
+
+## Unresolved questions
+
+- Should we provide an automated transformation for people using the existing schemas? How should we provide this transformation: as a new API in `cedar-policy`, or a standalone executable? How long should we maintain the transformation code? How can we ensure that users run the transformation at upgrade time?
```

Comment:
Original discussion: https://github.com/cedar-policy/rfcs/pull/35#discussion_r1378832191

Reply:
My personal preference is that we not provide this. I'm worried that we're signing up for extra code to maintain that no one will use.